Add a flag to allow use of unverified SSL certs on Python 2.7.9+

nikvdp opened the pull request:
In Python 2.7. the behavior of `httplib` was changed to verify SSL certs by default (see [this](https://docs.python.org/2/library/httplib.html#httplib.HTTPSConnection)), making couchdb-python unusable with CouchDB servers using self-signed SSL certificates. 

These commits add an extra parameter to the `Server()` class optionally disabling SSL verification to re-allow the previous behavior.

kxepal commented:
I would not let leak low level protocol feature out to CouchDB API: you'd covered Server constructor, but not Database, Document and others which may be used directly by-passing Server instance. Especially, when leaks the feature which is limited to some very few Python versions. Given that fact, I think there is need to keep changes for http.py and let users do the following:

```
db = Database(url)
db.resource.session.disable_ssl_verification = True
```

In their code if they need in such behaviour. What do you think?

nikvdp commented:
Makes sense. Only problem is that the ConnectionPool needs to be re-initialized to use the InsuecureHTTPSConnection class. I've reverted the constructor changes and added a `disable_ssl_verification()` method to the Session class, which re-initializes the ConnectionPool correctly. 

So now to use the user should do: 
```
db = Database(url)
db.resource.session.disable_ssl_verification()
```

That work?

Also with this approach I think we had better add an example of how to do this to the documentation as this  is quite non-obvious for someone who is just trying to use the library.

kxepal commented:
Indeed, ConnectionPool need to be reinitialized. Hopefully that's one-time operation. LGFM, except lines are need to fit 80-chars right margin (:

+1 to add few notes about in docs/docstrings.

nikvdp commented:
OK, made the changes.

djc commented:
@nikvdp, thanks for you work on this! Also thanks to @kxepal for the reviews, you had very much the same remarks as I had when I first saw this work (when @nikvdp forked the project).

I've applied a squashed version of your patches in 43ee3891 and tightened up some formatting in 04ab8401.
